samcli/cli/types.py

```diff
@@ -232,6 +239,17 @@ def convert(self, value, param, ctx):
 
         return result
 
+    def _add_value(self, result, key, new_value):
+        """
+        Add a given value to a given key in the result map.
+        """
+        if self.multiple_values_per_key:
```

Comment:
Can this be deduced from the type?

Reply:
The values are always strings, `multiple_values_per_key` just determines whether or not to accumulate values in the same key or to replace them.

With `multiple_values_per_key=True`:
Key1=Value1 Key1=Value2 -> {Key1: [Value1, Value2]}

Without `multiple_values_per_key` specified (before this change):
Key1=Value1 Key1=Value2 -> {Key1: Value2}